Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -173,27 +184,37 @@ class Navigator {
     /// the vector of navigation surfaces to work through
     NavigationSurfaces navSurfaces = {};
     /// the current surface index of the navigation state
-    std::optional<std::size_t> navSurfaceIndex;
+    std::optional<std::size_t> navSurfaceIndex{std::nullopt};
 
     // Navigation on layer level
     /// the vector of navigation layers to work through
     NavigationLayers navLayers = {};
     /// the current layer index of the navigation state
-    std::optional<std::size_t> navLayerIndex;
+    std::optional<std::size_t> navLayerIndex{std::nullopt};
 
     // Navigation on volume level
     /// the vector of boundary surfaces to work through
     NavigationBoundaries navBoundaries = {};
     /// the current boundary index of the navigation state
-    std::optional<std::size_t> navBoundaryIndex;
+    std::optional<std::size_t> navBoundaryIndex{std::nullopt};
 
     // Navigation candidates(portals and surfaces together)
     /// the vector of navigation candidates to work through
     NavigationCandidates navCandidates = {};
     /// the current candidate index of the navigation state
-    std::optional<std::size_t> navCandidateIndex;
-
+    std::optional<std::size_t> navCandidateIndex{std::nullopt};
+
+    // Navigation on free surface level. Free surfaces are traversed
+    NavigationCandidates freeCandidates = {};
```

Comment:
So we need a separate candidate set? Is there a way to pre-populate the candidates object that we're about to hand over to the policy? That way, they would end up in the same candidate list, and you would have to do less book keeping later on, no?

Reply:
Ok, so this is now only used to hold the candidates after they've been copied over from the options struct?